numba/core/runtime/nrt_external.h

```diff
@@ -7,6 +7,18 @@ typedef struct MemInfo NRT_MemInfo;
 
 typedef void NRT_managed_dtor(void *data);
 
+typedef void *(*NRT_external_malloc_func)(size_t size, void *opaque_data);
+typedef void *(*NRT_external_realloc_func)(void *ptr, size_t new_size, void *opaque_data);
+typedef void (*NRT_external_free_func)(void *ptr, void *opaque_data);
+
+struct ExternalMemAllocator {
+    NRT_external_malloc_func malloc;
+    NRT_external_realloc_func realloc;
+    NRT_external_free_func free;
+    void *opaque_data;
```

Comment:
As discussed, the `opaque_data` needs to be a function pointer to `DPCTLQmgr_GetCurrentQueue` so that the external allocator always uses the current queue from `dpctl` for memory allocation. The current implementation stores the queue only once and after that it is never updated, leading to potential issues if the current queue is changed inside `dpctl`.

Reply:
The spot to mention this would be in dppl_rt.c where we fill in the opaque value.  This here is still fine since opaque_data can also store a function pointer.